### PR TITLE
make discoverySelector work on gateway deployment controller

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

Fix https://github.com/istio/istio/issues/45137

### CODE REVIEW

**Review by keithmattix — COMMENTED** [review]
LGTM; leaving for other reviews though

**Review by linsun — COMMENTED** [review]
LGTM

Can you pls add a release note and clarify if this is for gateway deploy controller for the Istio gateway class only? what about istio-waypoint?

File: pilot/pkg/config/kube/gateway/deploymentcontroller.go [hunk 1]
```diff
@@ -164,8 +165,13 @@ func getClassInfos() map[gateway.GatewayController]classInfo {
 // The controller will not start until Run() is called.
 func NewDeploymentController(client kube.Client, clusterID cluster.ID, env *model.Environment,
 	webhookConfig func() inject.WebhookConfig, injectionHandler func(fn func()), tw revisions.TagWatcher, revision string,
+	nsFilter namespace.DiscoveryNamespacesFilter,
 ) *DeploymentController {
-	gateways := kclient.New[*gateway.Gateway](client)
+	var filter namespace.DiscoveryFilter
```
Comment (howardjohn) [inline pilot/pkg/config/kube/gateway/deploymentcontroller.go:170]: please gate by EnableEnhancedResourceScoping

Comment (hzxuzhonghu) [reply pilot/pkg/config/kube/gateway/deploymentcontroller.go:170]: good catch, will add the caller

**Review by howardjohn — APPROVED** [review]

**Pull request merged**